### PR TITLE
Offline/Online Branchformer Transducer

### PR DESCRIPTION
**Pull request by b-flo**
Hi,

This PR add Branchformer block with offline and streaming capabilities to the new Transducer version.
Btw, I need to redesign the units tests in the future, it'll become a mess if I pile up block tests. It's OK for now because the other block types aren't a priority.

Also, as follow up to #4479:

- Set the conv mod normalization module for Conformer to `BatchNorm1d`. Subsequently, the type was removed from `get_normalization(...)`.
- Fix some main and Conformer parameters in the ESPnet2 tutorial doc.
- Moved the dynamic chunk training parameters description from Encoder section to Streaming section.

### CODE REVIEW

**Comment by b-flo** [comment]
@pyf98 Could you review this PR when you have time please? I can't assign you it seems.

**Comment by pyf98** [comment]
Thanks @b-flo , I'll read this PR.

**Comment by b-flo** [comment]
Do not mind the changes from the last commit, it's only cosmetic (docs + typos). I needed to squeeze changes from ext. review into this branch to do some reorganization/rebase.

**Comment by sw005320** [comment]
@pyf98, can you review it?

File: espnet2/asr_transducer/encoder/modules/convolution.py [hunk 1]
```diff
@@ -107,3 +104,95 @@ def forward(
...
+        dropout_rate: Dropout rate.
+        causal: Whether to use causal convolution (set to True if streaming).
+
+    """
+
+    def __init__(
+        self,
+        size: int,
+        kernel_size: int,
+        activation: torch.nn.Module = torch.nn.ReLU(),
```
Comment (pyf98) [inline espnet2/asr_transducer/encoder/modules/convolution.py:127]: For CSGU, I recommend to set the `activation` as `Identity()`. Based on previous experiments, a nonlinear activation does not improve the performance.

**Comment by pyf98** [comment]
Thanks @b-flo! This PR looks good to me.

I haven't tested the code though. We may test Branchformer or its variant in the Transducer framework later.

File: espnet2/asr_transducer/encoder/modules/convolution.py (hunk 1 above)
Comment (b-flo) [reply espnet2/asr_transducer/encoder/modules/convolution.py]: Removed activation argument, I manually set `Identity()` for now.

**Comment by b-flo** [comment]
I think I'll split `main_params` config into dedicated shared configs in the future (e.g.: for attention, norm, activation, etc). It'll be easier to set proper parameters while keeping flexibility and code clean. It'll be better for delegating some parts to each block type  for example.
Not a priority right now though.

> I haven't tested the code though. We may test Branchformer (or its variant) in the Transducer framework later.

I trained an offline and online model on Librispeech-100. I didn't have time/resources to tune it so I couldn't reach Conformer perf but it seemed to work correctly.

**Comment by b-flo** [comment]
@sw005320 I'll merge this PR, is that OK for you?

**Comment by sw005320** [comment]
LGTM!
Thanks, @b-flo!
